File: engine/src/main/java/de/hanno/hpengine/engine/graphics/shader/ComputeShader.java
package de.hanno.hpengine.engine.graphics.shader;

import de.hanno.hpengine.engine.graphics.shader.define.Defines;
import de.hanno.hpengine.util.ressources.CodeSource;

public class ComputeShader extends AbstractShader {

    public static ComputeShader load(ProgramManager programManager, CodeSource sourceCode) {
        return load(programManager, sourceCode, new Defines());
    }

    public static ComputeShader load(ProgramManager programManager, CodeSource sourceCode, Defines defines) {
        return programManager.loadShader(ComputeShader.class, sourceCode, defines);
    }

    @Override
    public OpenGLShader getShaderType() {
        return OpenGLShader.ComputeShader;
    }
}

File: engine/src/main/java/de/hanno/hpengine/engine/graphics/shader/ComputeShader.kt
package de.hanno.hpengine.engine.graphics.shader

import de.hanno.hpengine.engine.graphics.shader.Shader.ShaderType
import de.hanno.hpengine.engine.graphics.shader.define.Defines
import de.hanno.hpengine.util.ressources.CodeSource
import de.hanno.hpengine.util.ressources.FileBasedCodeSource

class ComputeShader(override var shaderSource: CodeSource, override var id: Int = -1) : Shader {
    override val shaderType: ShaderType = ShaderType.ComputeShader

    companion object {
        @JvmOverloads
        fun load(programManager: OpenGlProgramManager, sourceCode: FileBasedCodeSource, defines: Defines = Defines()): ComputeShader {
            return ComputeShader(sourceCode, programManager.loadShader(ShaderType.ComputeShader, sourceCode, defines))
        }
    }
}